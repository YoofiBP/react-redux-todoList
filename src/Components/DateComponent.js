import React from 'react';
import IconButton from "@material-ui/core/IconButton";
import ArrowForwardIosIcon from '@material-ui/icons/ArrowForwardIos';
import ArrowBackIosIcon from '@material-ui/icons/ArrowBackIos';
import dayjs from 'dayjs'
import { changeDate } from '../redux/actions';
import { connect } from 'react-redux';
import { DATE_BACK, DATE_FORWARD } from '../redux/actionTypes';

function DateComponent(props){
    let today = (props.currentDate === dayjs().format("MM/DD/YYYY")) ? "Today" : dayjs(props.currentDate).format('ddd, MMM D, YYYY')
    return (
        <>
        <IconButton
        aria-label="delete"
        onClick={() => props.changeDate(DATE_BACK)}
        edge="end"
        className="p-2"
        size="small"
        id="back"
      >
        <ArrowBackIosIcon />
      </IconButton>
      <h1>{today}</h1>
      <IconButton
        aria-label="delete"
        onClick={() => props.changeDate(DATE_FORWARD)}
        edge="end"
        className="p-2"
        size='small'
        id="forward"
      >
        <ArrowForwardIosIcon />
      </IconButton>
      </>
    )
}

const mapStateToProps = (state) => {
  return {currentDate: state.date }
}

const mapDispatchToProps = dispatch => {
  return {
    changeDate: direction => {
      dispatch(changeDate(direction));
    }
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(DateComponent);