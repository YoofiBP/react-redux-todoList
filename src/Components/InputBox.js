import React from 'react';
import { connect } from 'react-redux';
import { addItem } from '../redux/actions';
//import { users } from '../users';

class InputBox extends React.Component {
    constructor(props){
        super(props);
        this.state = {
            value:""
        }
        this.handleChange = this.handleChange.bind(this)
        this.handleSubmit = this.handleSubmit.bind(this)
    }

    handleChange(event){
        this.setState({value: event.target.value})
    }

    handleSubmit(event){
        if(this.state.value.length > 1){
        this.props.addNewTodo(this.state.value, this.props.currentDate)
        this.setState({value:""})
        }
        event.preventDefault();
    }

    render(){
    return (
        <div className="input-group mb-2">
            <form className="input-group mb-2 w-100" onSubmit={this.handleSubmit}>
  <input id="title" onChange={this.handleChange} type="text" className="form-control" value={this.state.value} placeholder="New Item" aria-label="New Item" aria-describedby="button-addon2"/>
  <input type="submit" className="btn btn-outline-secondary" value="Add Item" />
  </form>
</div>
    )
    }
}

export class Welcome extends React.Component{
    constructor(props){
        super(props);
        this.state = {user: {}}
    }

    componentDidMount() {
    fetch('/users.json')
    .then(res => res.json())
    .then(users => { return users.filter(user => user.userId === this.props.userId)})
    .then(user => this.setState({user: user[0]}))
    .catch(err => console.log(err))
    }

    render() {
        return (
            <h1>Welcome {this.state.user.username}!</h1>
        )
    }
    
}

const mapDispatchToProps = dispatch => {
    return {
        addNewTodo: (title, date) => {
            dispatch(addItem(title, date))
        }
    }
}

const mapStateToProps = state => {
    return {
        currentDate: state.date
    }
}   

export default connect(mapStateToProps, mapDispatchToProps)(InputBox);