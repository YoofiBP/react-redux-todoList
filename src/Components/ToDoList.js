import React from 'react';
import TodoListItem from './ToDoListItem'
import { connect } from 'react-redux'

function ToDoList(props) {
    return (
        props.items.map(item => <TodoListItem item={item} key={item.id} id={item.id}/>)
    )
}

const mapStateToProps = state => {
    let [todaysItems] = state.todoList.filter(day => day.date === state.date)
    return todaysItems ? { items: todaysItems.tasks } : { items: [] }
}

export default connect(mapStateToProps)(ToDoList);