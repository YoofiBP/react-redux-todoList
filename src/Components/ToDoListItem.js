import React from "react";
import IconButton from "@material-ui/core/IconButton";
import DeleteIcon from "@material-ui/icons/Delete";
import InputIcon from "@material-ui/icons/Input";
import dayjs from "dayjs";
import Tooltip from "@material-ui/core/Tooltip";

import { connect } from 'react-redux';
import { addItem, checkItem, deleteItem } from '../redux/actions';

function TodoListItem(props) {
  const { title, completed } = props.item;
  return (
    <div className="d-flex">
      <li>
        <input
          type="checkbox"
          value={title}
          checked={completed}
          className="mr-auto p-2"
          onChange={() => props.checkItem(props.id,props.currentDate)}
        />{" "}
        <label
          className="mr-auto p-2"
          style={
            completed
              ? { textDecoration: "line-through", color: "grey" }
              : { textDecoration: "none" }
          }
        >
          {title}
        </label>
        <Tooltip title="Delete">
          <IconButton
            aria-label="delete"
            onClick={() => props.deleteItem(props.id, props.currentDate)}
            edge="end"
            className="p-2"
            id="delete"
          >
            <DeleteIcon />
          </IconButton>
        </Tooltip>
        <Tooltip title="Do Tomorrow">
          <IconButton onClick={() => props.moveItem(props.id, props.currentDate, title)} id="move">
            <InputIcon />
          </IconButton>
        </Tooltip>
      </li>
    </div>
  );
}

const mapDispatchToProps = dispatch => {
  return {
    checkItem: (index, date) => {
      dispatch(checkItem(index, date))
    },
    deleteItem: (index, date) => {
      dispatch(deleteItem(index, date))
    },
    moveItem: (index, date, title) => {
      dispatch(deleteItem(index, date));
      date = dayjs(date, "MM/DD/YYYY").add(1, "day").format("MM/DD/YYYY")
      dispatch(addItem(title, date))
    }
  }
}

const mapStateToProps = state => {
  return {
      currentDate: state.date
  }
} 
export default connect(mapStateToProps, mapDispatchToProps)(TodoListItem);