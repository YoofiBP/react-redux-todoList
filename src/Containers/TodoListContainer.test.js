//import { unmountComponentAtNode, render } from 'react-dom';
import React from "react";
import { shallow } from 'enzyme'
import { TodoListItem } from '../Components/ToDoListItem';
import { ToDoList } from '../Components/ToDoList';
import { create } from 'react-test-renderer';

/**
 Testing units
 Rendering
 User events (Clicking, typing, checking)

 */

 const testItem = {
     id: 1,
     title: "Learn React",
     completed: false
 }

 const testItem2 = {
   id:2,
   title: "Learn React Part 2",
   completed: true
 }

 const todos =  [
        {
          userId: 1,
          id: 1,
          title: "Learn React",
          completed: false
        },
        {
          userId: 1,
          id: 2,
          title: "Conduct Internal VAPT",
          completed: false
        },
        {
          userId: 1,
          id: 3,
          title: "Physical Security",
          completed: false
        },
        {
          userId: 1,
          id: 4,
          title: "Application Security",
          completed: false
        }
      ]

describe('Testing todoList', () => {
    it('Shallow renders without crashing', () => {
        shallow(<ToDoList items={todos}/>)
    })

    it('ToDoList matches snapshot', () => {
        const todoList = create(<ToDoList items={todos} />).toJSON();
        expect(todoList).toMatchSnapshot();
    })

    it('Renders the correct number of todoListItem Components', () => {
      const wrapper = shallow(<ToDoList items={todos}/>)
      expect(wrapper.find(TodoListItem)).toHaveLength(4);
    })
})

describe('Testing todoListItem', () =>{
    it('Shallow renders without crashing', () => {
        shallow(<TodoListItem item={testItem} key={testItem.id} id={testItem.id} />)
    })
   
    it('TodoListItem matches a snapshot', () => {
       const todoListItem = create(<TodoListItem item={testItem} key={testItem.id} id={testItem.id} />).toJSON()
       expect(todoListItem).toMatchSnapshot();
    })

    it('Appropriate props are passed from parent', () => {
      let onClick, onChange, onMove;
      const todoItems = todos;
      onClick = onChange = onMove = jest.fn();
      const wrapper = shallow(<ToDoList items={todoItems} onClick={onClick} onChange={onChange} onMove={onMove}/>);
      const toDoListItemWrapper = wrapper.find(TodoListItem);
      toDoListItemWrapper.forEach((todoListItem) => {
        expect(todoListItem.props()).toEqual(expect.objectContaining({
          item: expect.any(Object),
          id: expect.any(Number),
          onClick: onClick,
          onChange: onChange,
          onMove: onMove
        }))
      })
    })

    it('has appropriate number of child components', () => {
      const wrapper = shallow(<TodoListItem item={testItem} key={testItem.id} id={testItem.id}/>);
      expect(wrapper.find('li')).toHaveLength(1);
      expect(wrapper.find('#delete')).toHaveLength(1);
      expect(wrapper.find('#move')).toHaveLength(1);
    })
    
    it('Passes appropriate props', () => {
      const onClick = jest.fn();
      const onMove = jest.fn();
      const onChange = jest.fn();
      const wrapper = shallow(<TodoListItem item={testItem} key={testItem.id} id={testItem.id} onClick={onClick} onChange={onChange} onMove={onMove}/>);
      expect(wrapper.find('label').text()).toEqual('Learn React');
      expect(wrapper.find('label').prop('style')).toEqual({textDecoration: "none"})
      expect(wrapper.find('input').props()).toEqual(expect.objectContaining({
        type: "checkbox",
        value: "Learn React",
        checked: false,
        onChange: expect.any(Function)
      }))
      expect(wrapper.find('#delete').prop('onClick')).toEqual(expect.any(Function))
      expect(wrapper.find("#move").prop('onClick')).toEqual(expect.any(Function))
    })

    it("Prop functions are called when events are triggered", () => {
      const onClick = jest.fn();
      const onMove = jest.fn();
      const onChange = jest.fn();
      const wrapper = shallow(<TodoListItem item={testItem2} key={testItem2.id} id={testItem.id} onClick={onClick} onChange={onChange} onMove={onMove}/>);
      const itemCheckBox = wrapper.find('input');
      const deleteButton = wrapper.find('#delete');
      const moveButton = wrapper.find('#move');
      itemCheckBox.prop('onChange')();
      expect(onChange).toHaveBeenCalled();
      expect(wrapper.find('label').prop('style')).toEqual({ textDecoration: "line-through", color: "grey" });
      deleteButton.prop('onClick')();
      expect(onClick).toHaveBeenCalled();
      moveButton.prop('onClick')();
      expect(onMove).toHaveBeenCalled();
    })
})
 