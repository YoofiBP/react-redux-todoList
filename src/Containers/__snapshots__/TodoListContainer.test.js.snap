// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing todoList ToDoList matches snapshot 1`] = `
Array [
  <div
    className="d-flex"
  >
    <li>
      <input
        checked={false}
        className="mr-auto p-2"
        onChange={[Function]}
        type="checkbox"
        value="Learn React"
      />
       
      <label
        className="mr-auto p-2"
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        Learn React
      </label>
      <button
        aria-describedby={null}
        aria-label="delete"
        className="MuiButtonBase-root MuiIconButton-root p-2 MuiIconButton-edgeEnd"
        disabled={false}
        id="delete"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        title="Delete"
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
      <button
        aria-describedby={null}
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        id="move"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        title="Do Tomorrow"
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M21 3.01H3c-1.1 0-2 .9-2 2V9h2V4.99h18v14.03H3V15H1v4.01c0 1.1.9 1.98 2 1.98h18c1.1 0 2-.88 2-1.98v-14c0-1.11-.9-2-2-2zM11 16l4-4-4-4v3H1v2h10v3z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </li>
  </div>,
  <div
    className="d-flex"
  >
    <li>
      <input
        checked={false}
        className="mr-auto p-2"
        onChange={[Function]}
        type="checkbox"
        value="Conduct Internal VAPT"
      />
       
      <label
        className="mr-auto p-2"
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        Conduct Internal VAPT
      </label>
      <button
        aria-describedby={null}
        aria-label="delete"
        className="MuiButtonBase-root MuiIconButton-root p-2 MuiIconButton-edgeEnd"
        disabled={false}
        id="delete"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        title="Delete"
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
      <button
        aria-describedby={null}
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        id="move"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        title="Do Tomorrow"
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M21 3.01H3c-1.1 0-2 .9-2 2V9h2V4.99h18v14.03H3V15H1v4.01c0 1.1.9 1.98 2 1.98h18c1.1 0 2-.88 2-1.98v-14c0-1.11-.9-2-2-2zM11 16l4-4-4-4v3H1v2h10v3z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </li>
  </div>,
  <div
    className="d-flex"
  >
    <li>
      <input
        checked={false}
        className="mr-auto p-2"
        onChange={[Function]}
        type="checkbox"
        value="Physical Security"
      />
       
      <label
        className="mr-auto p-2"
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        Physical Security
      </label>
      <button
        aria-describedby={null}
        aria-label="delete"
        className="MuiButtonBase-root MuiIconButton-root p-2 MuiIconButton-edgeEnd"
        disabled={false}
        id="delete"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        title="Delete"
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
      <button
        aria-describedby={null}
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        id="move"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        title="Do Tomorrow"
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M21 3.01H3c-1.1 0-2 .9-2 2V9h2V4.99h18v14.03H3V15H1v4.01c0 1.1.9 1.98 2 1.98h18c1.1 0 2-.88 2-1.98v-14c0-1.11-.9-2-2-2zM11 16l4-4-4-4v3H1v2h10v3z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </li>
  </div>,
  <div
    className="d-flex"
  >
    <li>
      <input
        checked={false}
        className="mr-auto p-2"
        onChange={[Function]}
        type="checkbox"
        value="Application Security"
      />
       
      <label
        className="mr-auto p-2"
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        Application Security
      </label>
      <button
        aria-describedby={null}
        aria-label="delete"
        className="MuiButtonBase-root MuiIconButton-root p-2 MuiIconButton-edgeEnd"
        disabled={false}
        id="delete"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        title="Delete"
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
      <button
        aria-describedby={null}
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        id="move"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        title="Do Tomorrow"
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M21 3.01H3c-1.1 0-2 .9-2 2V9h2V4.99h18v14.03H3V15H1v4.01c0 1.1.9 1.98 2 1.98h18c1.1 0 2-.88 2-1.98v-14c0-1.11-.9-2-2-2zM11 16l4-4-4-4v3H1v2h10v3z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </li>
  </div>,
]
`;

exports[`Testing todoListItem TodoListItem matches a snapshot 1`] = `
<div
  className="d-flex"
>
  <li>
    <input
      checked={false}
      className="mr-auto p-2"
      onChange={[Function]}
      type="checkbox"
      value="Learn React"
    />
     
    <label
      className="mr-auto p-2"
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      Learn React
    </label>
    <button
      aria-describedby={null}
      aria-label="delete"
      className="MuiButtonBase-root MuiIconButton-root p-2 MuiIconButton-edgeEnd"
      disabled={false}
      id="delete"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      title="Delete"
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
    <button
      aria-describedby={null}
      className="MuiButtonBase-root MuiIconButton-root"
      disabled={false}
      id="move"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      title="Do Tomorrow"
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M21 3.01H3c-1.1 0-2 .9-2 2V9h2V4.99h18v14.03H3V15H1v4.01c0 1.1.9 1.98 2 1.98h18c1.1 0 2-.88 2-1.98v-14c0-1.11-.9-2-2-2zM11 16l4-4-4-4v3H1v2h10v3z"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </li>
</div>
`;
